### DISCUSSION TITLE
Cannot create a package from a library managed with Cmake

### DISCUSSION THREAD

**Discussion by hugoc7** [xmake-io/xmake#2103] [Q&A] [answered]
Hi! I'm very glad to use Xmake and it's great, thank you for your work ! However, I am struggling to create a Xmake package from a small library managed with Cmake : [https://github.com/svperbeast/struct](https://github.com/svperbeast/struct). Here is my lua script : 
```lua
package("struct")
    set_description("pack and unpack binary data.")
    set_license("MIT-License")
    set_homepage("https://github.com/svperbeast/struct")
    --add_urls("https://github.com/svperbeast/struct.git")

    add_deps("cmake")
    set_sourcedir(path.join(os.scriptdir(), "struct"))
    on_install(function (package)
       import("package.tools.cmake").install(package, {"-DSTRUCT_BUILD_TEST=OFF"})
    end)

    on_test(function (package)
        assert(package:has_cfuncs("struct_pack", {includes = "struct.h"}))
        assert(package:has_cfuncs("struct_unpack", {includes = "struct/struct.h"}))
    end)
package_end()
```
When I run xmake I have the following problem : `C:\Users\Hugo\AppData\Local\Temp\.xmake\220226\_48C38E74F9CB4E10834A530A09989430.c(2): fatal error C1083: Unable to open include file : 'struct/struct.h' : No such file or directory`. 

I noticed that I get the correct library and header files in :
`${projectdir}\build\.packages\s\struct\latest\cache\build_8574311f\Release\include`
and
`${projectdir}\build\.packages\s\struct\latest\cache\build_8574311f\Release\lib`
So the library is correctly build but it seems that xmake can't find the output lib and headers files.

Thanks for your help.

**Comment by waruqi** [comment, marked as answer]
The cmakelists of this project are not well implemented cmake install, but you can copy them yourself to package:installdir("include")

```lua
        import("package.tools.cmake").build(package, configs, {buildir = "build"})
        os.cp("build/release/lib", package:installdir())
        os.cp("build/release/include", package:installdir())
```

**Reply by hugoc7** [reply]
Thanks a lot for your quick answer, now my package works fine on my computer!

 I didn't find any information about **package.tools.cmake** in the manual, but I looked at the xmake github repo ([https://github.com/xmake-io/xmake/blob/master/xmake/modules/package/tools/cmake.lua](https://github.com/xmake-io/xmake/blob/master/xmake/modules/package/tools/cmake.lua)) to see the options availables

**Reply by xq114** [reply]
There are many existing examples in https://github.com/xmake-io/xmake-repo , have a look and I think that's what you need
